lib/resolve.ts

```diff
@@ -104,3 +111,23 @@ export function resolveTask(
 
 	return task;
 }
+
+/**
+ * Given a balena architecture string, translate it to the equivalent
+ * docker platform string.
+ */
+function resolveDockerPlatform(balenaArchitecture: string): string | null {
+	switch (balenaArchitecture) {
+		case 'amd64':
+			return 'linux/amd64';
+		case 'i386':
+			return 'linux/386';
+		case 'aarch64':
+			return 'linux/arm64';
+		case 'armv7hf':
+			return 'linux/arm/v7';
+		case 'rpi':
+			return 'linux/arm/v6';
+	}
+	return null;
```

Comment:
What about using an empty string instead of `null`? The "advantage" being that the return type is simpler as `string` instead of `string | null`. An empty string is a [falsy value](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), and could be tested with:
```
	if (platform) {
		task.dockerPlatform = platform;
	}
```
Or on a single line:
```
	task.dockerPlatform = platform || task.dockerPlatform;
```

Reply:
:ok_hand: